app/models/submission.rb

```diff
@@ -32,5 +34,14 @@ def reject
   def average_rate
     rates.count == 0 ? 0 : (rates.sum(:value).to_f / rates.count)
   end
-end
 
+  def generate_confirmation_token!
+    self.confirmation_token = Devise.friendly_token
+    self.confirmation_token_created_at = Time.now
+    save
+  end
+
+  def has_expired?
```

Comment:
I would rename this method, because it doesn't tell me what is expired - invitation or submission?